test/test_resource.rs

```diff
@@ -2,19 +2,17 @@ use nix::sys::resource::{Resource, getrlimit, setrlimit};
 
 #[test]
 pub fn test_resource_limits() {
-    let mut limit = getrlimit(Resource::RLIMIT_STACK).unwrap();
-    assert!(limit.0 != limit.1);
+    let (mut soft_limit, hard_limit) = getrlimit(Resource::RLIMIT_STACK).unwrap();
+    let orig_limit = (soft_limit, hard_limit);
```

Comment:
Your changes to the test do fix the previous failure, but the coverage isn't very good, since the soft and hard limits are often identical.  How about setting `RLIMIT_NOFILE` to a hard-coded value of 1024 if the current soft limit is unlimited, or the current softlimit - 1 otherwise?

Reply:
Done

Reply:
Do we really need two tests?  Does the stack test provide any extra coverage over the `RLIMIT_NOFILE` test?